src/System.Private.ServiceModel/src/System/ServiceModel/Security/ClientCredentialsSecurityTokenManager.cs

```diff
@@ -70,89 +70,9 @@ internal protected bool IsIssuedSecurityTokenRequirement(SecurityTokenRequiremen
 
         public override SecurityTokenProvider CreateSecurityTokenProvider(SecurityTokenRequirement tokenRequirement)
         {
-            return this.CreateSecurityTokenProvider(tokenRequirement, false);
+            throw ExceptionHelper.PlatformNotSupported("CreateSecurityTokenProvider(SecurityTokenRequirement tokenRequirement) not supported");
```

Comment:
No new scenarios will hit a PNSE. In the original code, every code path that could be taken to get a TokenProvider always resulted in a PNSE. This changes removes all the checks to decide which line of code would be the lucky winner to throw a PNSE. The other methods in this class all have the exact same pattern of throwing PNSE with this format of message. We only need a TokenProvider with message security. If/when we implement that, this code will end up looking quite different anyway.